nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/storage/FetchGCSObject.java

```diff
@@ -233,8 +233,15 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
                 throw new StorageException(404, "Blob " + blobId + " not found");
             }
 
+            if (rangeStart > 0 && rangeStart >= blob.getSize()) {
+                getLogger().debug("Start position: {}, blob size: {}", new Object[] {rangeStart, blob.getSize()});
```

Comment:
Minor thing, in general we try to wrap the debugs in an `isDebugEnabled()`. This one should (hopefully) be reached only very rarely so definitely not a big deal, but thought I'd mention it for the future.
```suggestion
                if (getLogger().isDebugEnabled() {
                    getLogger().debug("Start position: {}, blob size: {}", new Object[] {rangeStart, blob.getSize()});
                }
```

Reply:
Thank you, that makes sense.